lib/items/facebook.dart

```diff
@@ -1,52 +1,62 @@
 import 'package:url_launcher/url_launcher.dart';
+import 'package:http/http.dart' as http;
+import 'package:html/parser.dart' show parse;
 
 import 'item.dart';
 
 class FacebookItem extends MultaccItem {
-  String url;
-  String pageName;
+  String username;
+  String userId;
 
   FacebookItem();
 
   FacebookItem.fromJson(Map<String, dynamic> json)
-      : url = json['url'],
-        pageName = json['page'];
+      : username = json['username'],
+        userId = json['id'];
 
-  toMap() => {'url': url, 'page': pageName};
+  toMap() => {'username': username, 'userId': userId};
 
-  get humanReadableValue => 'fb/$pageName';
+  get humanReadableValue => username ?? (userId != null ? 'fb.com/$userId' : '');
 
   get type => MultaccItemType.Facebook;
 
   launchApp() async {
     try {
-      await launch('fb://profile/$pageName');
+      await launch('fb://profile/$userId');
     } catch (ex) {
-      launch(url);
+      launch('https://facebook.com/${username ?? userId}');
     }
   }
 
   launchMessenger() async {
     try {
-      await launch('fb-messenger://user/$pageName');
+      await launch('fb-messenger://user/${username ?? userId}');
     } catch (ex) {
-      launch('https://facebook.com/messages/t/$pageName');
+      launch('https://facebook.com/messages/t/${username ?? userId}');
     }
   }
 
   get isLaunchable => true;
 
   set value(String input) {
-    url = input;
-    input = input.toLowerCase();
-    if (input.contains('fb.me') || input.contains('m.me')) {
-      pageName = input.split('me/')[1];
+    String domain = input.toLowerCase().split('/')[0];
+
+    if (domain.contains('fb.me') || domain.contains('m.me')) {
+      username = input.split('/').last;
     } else if (input.contains('profile.php?id=')) {
-      pageName = input.split('id=')[1];
-    } else if (input.contains('facebook.com/')) {
-      pageName = input.split('.com/')[1];
+      userId = input.split('id=')[1];
+    } else if (domain.contains('facebook.com') || domain.contains('fb.com')) {
+      username = input.split('/').last;
```

Comment:
won't work if there is other stuff after id or username (such as / or # or some tracking params)

```suggestion
      userId = input.split('id=')[1];
      userId = RegExp(r'[&?]id=(\d+)').firstMatch(lowerCaseInput).group(1)
    } else if (lowerCaseInput.contains('facebook.com/') || lowerCaseInput.contains('fb.com/')) {
      username = RegExp(r'\.com/([a-zA-Z0-9.]{5,})').firstMatch(input).group(1);
```